agent/stats/task_windows.go

```diff
@@ -0,0 +1,179 @@
+	}
+	str := string(out)
+
+	lines := strings.Split(str, "\n")
+	m := make(map[string]string)
+	for _, line := range lines {
+		// populate all the network metrics in a map
+		kv := strings.Split(line, ":")
+		if len(kv) != 2 {
+			continue
+		}
+		key := strings.TrimSpace(kv[0])
+		value := strings.TrimSpace(kv[1])
+		m[key] = value
+	}
+
+	receivedBroadcastPackets, err := strconv.ParseUint(m["ReceivedBroadcastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedBroadcastPackets", device)
+	}
+
+	receivedMulticastPackets, err := strconv.ParseUint(m["ReceivedMulticastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedMulticastPackets", device)
+	}
+
+	receivedUnicastPackets, err := strconv.ParseUint(m["ReceivedUnicastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedUnicastPackets", device)
+	}
+
+	sentBroadcastPackets, err := strconv.ParseUint(m["SentBroadcastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : SentBroadcastPackets", device)
+	}
+
+	sentMulticastPackets, err := strconv.ParseUint(m["SentMulticastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : field SentMulticastPackets", device)
+	}
+
+	sentUnicastPackets, err := strconv.ParseUint(m["SentUnicastPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : field SentUnicastPackets", device)
+	}
+
+	receivedBytes, err := strconv.ParseUint(m["ReceivedBytes"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedBytes", device)
+	}
+
+	receivedPacketErrors, err := strconv.ParseUint(m["ReceivedPacketErrors"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedPacketErrors", device)
+	}
+
+	receivedDiscardedPackets, err := strconv.ParseUint(m["ReceivedDiscardedPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : ReceivedDiscardedPackets", device)
+	}
+
+	sentBytes, err := strconv.ParseUint(m["SentBytes"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : SentBytes", device)
+	}
+
+	outboundPacketErrors, err := strconv.ParseUint(m["OutboundPacketErrors"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : OutboundPacketErrors", device)
+	}
+
+	outboundDiscardedPackets, err := strconv.ParseUint(m["OutboundDiscardedPackets"], 10, 64)
+	if err != nil {
+		return nil, errors.Errorf("failed to parse network stats for %s : OutboundDiscardedPackets", device)
+	}
+
+	numberOfContainers := uint64(taskStat.TaskMetadata.NumberContainers)
```

Comment:
Is there ever a case this could be 0? Should we do a safe check here or just allow the panic below?

Reply:
This value represents the number of containers present in the task and is not dynamically calculated. Any task would have atleast 1 container and that would be validated in MTS. Therefore, we should be fine here without any checks.

Reply:
Deal